src/link_manager/LinkManagerStateMachineActiveStandby.cpp

```diff
@@ -1158,10 +1174,14 @@ void ActiveStandbyStateMachine::LinkProberStandbyMuxUnknownLinkUpTransitionFunct
 {
     MUXLOGINFO(mMuxPortConfig.getPortName());
 
+    mStandbyUnknownUpCount++;
+
     if ((ps(mCompositeState) != ps(nextState)) &&
         (ps(nextState) == link_prober::LinkProberState::Label::Active ||
          ps(nextState) == link_prober::LinkProberState::Label::Standby)) {
         enterMuxWaitState(nextState);
+    } else if (mStandbyUnknownUpCount == mMuxPortConfig.getNegativeStateChangeRetryCount()) {
```

Comment:
Could this branch be missed if the code took the first if branch when condition is met?

Reply:
When it could, it should. Because there is a heartbeat state change here.

Reply:
Are you saying that the state change will reset the counter?

Reply:
It should have been reset. 

If it didn't get reset, the only case I can think of is user trigger a manual toggle in between, in that case, still we should skip the new branch.